### PR TITLE
ARROW-7830: [C++][Parquet] Use Arrow version number for parquet

### PR DESCRIPTION
**Pull request by emkornfield**

### CODE REVIEW

File: cpp/src/parquet/parquet_version.h.in [hunk 1]
```diff
@@ -18,14 +18,16 @@
...
+#define PARQUET_VERSION_MAJOR @ARROW_VERSION_MAJOR@
+#define PARQUET_VERSION_MINOR @ARROW_VERSION_MINOR@
+#define PARQUET_VERSION_PATCH @ARROW_VERSION_PATCH@
 
 #define PARQUET_SO_VERSION "@ARROW_SO_VERSION@"
 #define PARQUET_FULL_SO_VERSION "@ARROW_FULL_SO_VERSION@"
 
 // define the parquet created by version
-#define CREATED_BY_VERSION "parquet-cpp version @PARQUET_VERSION@"
+#define CREATED_BY_VERSION "parquet-cpp-arrow version @ARROW_VERSION@"
```
Comment (nevi-me) [inline cpp/src/parquet/parquet_version.h.in:31]: Is it useful to also add the git commit SHA? I can't tell if it gets added, from looking at the change

Comment (emkornfield) [reply cpp/src/parquet/parquet_version.h.in:31]: It isn't added here or in arrow.  Releases are tied to a git sha

Comment (emkornfield) [reply cpp/src/parquet/parquet_version.h.in:31]: not to say including a the sha hash isn't a good idea but I think it should be outside the scope of this work.

Comment (nevi-me) [reply cpp/src/parquet/parquet_version.h.in:31]: I understand, this change is already great, given how long the version hasn't changed

**Comment by pitrou** [comment]
@github-actions crossbow submit -g cpp

**Review by pitrou — APPROVED** [review]
+1

File: cpp/src/parquet/parquet_version.h.in [hunk 2]
```diff
@@ -18,14 +18,16 @@
 #ifndef PARQUET_VERSION_H
 #define PARQUET_VERSION_H
 
-#define PARQUET_VERSION_MAJOR @PARQUET_VERSION_MAJOR@
-#define PARQUET_VERSION_MINOR @PARQUET_VERSION_MINOR@
-#define PARQUET_VERSION_PATCH @PARQUET_VERSION_PATCH@
+#include "arrow/util/config.h"
```
Comment (kou) [inline cpp/src/parquet/parquet_version.h.in:21]: We don't need this because `@ARROW_*@` are expanded in build phase.

**Review by kou — COMMENTED** [review]
Could you also replace `@PARQUET_VERSION@` in `ParquetConfig.cmake.in`?

File: cpp/src/parquet/parquet_version.h.in (hunk 2 above)
Comment (emkornfield) [reply cpp/src/parquet/parquet_version.h.in]: removed.

**Comment by emkornfield** [comment]
> Could you also replace @PARQUET_VERSION@ in ParquetConfig.cmake.in?

I removed them is that OK, or do they need to be replaced?

**Comment by emkornfield** [comment]
> > Could you also replace @PARQUET_VERSION@ in ParquetConfig.cmake.in?
> 
> I removed them is that OK, or do they need to be replaced?

actually just replaced the parquet version assignment, let me know if that is what is needed.

**Review by kou — APPROVED** [review]
+1

I think that keeping `PARQUET_VERSION` in `ParquetConfig.cmake.in` is better for backward compatibility.

**Review by xhochy — APPROVED** [review]

**Pull request closed without merging**